### PR TITLE
CryptoStreamFactory method for decrypting InputStream

### PR DESCRIPTION
**Pull request by ellisjoe**

### CODE REVIEW

File: crypto-core/src/main/java/com/palantir/crypto2/io/CryptoStreamFactory.java [hunk 1]
```diff
@@ -98,4 +107,31 @@ private static OutputStream createDefaultEncryptedStream(OutputStream output, Ke
         return new CipherOutputStream(output, cipher.initCipher(Cipher.ENCRYPT_MODE));
     }
 
+    private static class StreamSeekableInput implements SeekableInput {
```
Comment (ellisjoe) [inline crypto-core/src/main/java/com/palantir/crypto2/io/CryptoStreamFactory.java:110]: could be replaced if this merges: https://github.com/palantir/seek-io/pull/15

File: crypto-core/src/main/java/com/palantir/crypto2/io/CryptoStreamFactory.java [hunk 2]
```diff
@@ -98,4 +107,31 @@ private static OutputStream createDefaultEncryptedStream(OutputStream output, Ke
...
+        private final InputStream input;
+
+        StreamSeekableInput(InputStream input) {
+            this.input = input;
+        }
+
+        @Override
+        public void seek(long offset) {
+            throw new UnsupportedOperationException();
```
Comment (uschi2000) [inline crypto-core/src/main/java/com/palantir/crypto2/io/CryptoStreamFactory.java:119]: who could possibly call this?

Comment (ellisjoe) [reply crypto-core/src/main/java/com/palantir/crypto2/io/CryptoStreamFactory.java:119]: it's impossible currently given we wrap the returned SeekableInput in a `SeekableInputStream` which doesn't expose `seek`

Comment (uschi2000) [reply crypto-core/src/main/java/com/palantir/crypto2/io/CryptoStreamFactory.java:119]: ok

File: crypto-core/src/main/java/com/palantir/crypto2/io/CryptoStreamFactory.java [hunk 3]
```diff
@@ -63,6 +64,14 @@ static SeekableInput decrypt(
         }
     }
 
+    /**
+     * Returns an {@link InputStream} that decrypts the given InputStream using the given {@link KeyMaterial} and
+     * cipher {@code algorithm}. When OpenSSL is available an implementation that uses AES-NI will be returned.
+     */
+    public static InputStream decrypt(InputStream input, KeyMaterial keyMaterial, String algorithm) {
```
Comment (uschi2000) [inline crypto-core/src/main/java/com/palantir/crypto2/io/CryptoStreamFactory.java:71]: would be good to add one sanity test

**Review by uschi2000 — APPROVED** [review]

File: crypto-core/src/main/java/com/palantir/crypto2/io/CryptoStreamFactory.java (hunk 3 above)
Comment (ellisjoe) [reply crypto-core/src/main/java/com/palantir/crypto2/io/CryptoStreamFactory.java:71]: done

**Review by uschi2000 — APPROVED** [review]

**Pull request merged**